evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+import pandas as pd
+from skopt.space import Integer
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils import deprecate_arg, import_or_raise
+
+
+class ARIMARegressor(Estimator):
+    """
+    Autoregressive Integrated Moving Average Model.
+    The three parameters (p, d, q) are the AR order, the degree of differencing, and the MA order.
+    More information here: https://www.statsmodels.org/devel/generated/statsmodels.tsa.arima_model.ARIMA.html
+
+    """
+    name = "ARIMA Regressor"
+    hyperparameter_ranges = {
+        "p": Integer(0, 10),
+        "d": Integer(0, 10),
+        "q": Integer(0, 10),
+    }
+    model_family = ModelFamily.ARIMA
+    supported_problem_types = [ProblemTypes.TIME_SERIES_REGRESSION]
+
+    def __init__(self, date_column=None, p=1, d=0, q=0,
```

Comment:
Can you please add `trend` as a parameter/hyperparameter? [Source](https://www.statsmodels.org/v0.12.0/generated/statsmodels.tsa.arima.model.ARIMA.html#statsmodels.tsa.arima.model.ARIMA) I think supporting the string values of n, c, t, and ct is enough for now. I think it would be great if this component was as self-contained as possible!